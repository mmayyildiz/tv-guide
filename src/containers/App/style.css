* {
    box-sizing: border-box;
}


li {
    display: list-item;
}

div {
    display: block;
}

img {
    max-width: 100%;
    height: auto;
    border: 0ppx;
}
