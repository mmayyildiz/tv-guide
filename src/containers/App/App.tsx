import React, { useEffect } from 'react';
import { connect } from 'react-redux';
import Channel from '../../models/Channel';
import { StoreState } from '../../store';
import TvGuide from '../TvGuide/TvGuide';
import { fetchChannels } from '../../store/channel/actions'
import './style.css'

interface AppProps {
    channels: Channel[];
    fetchChannels: Function
}

const App = ({ channels, fetchChannels }: AppProps) => {

    useEffect(() => {
        if (channels.length < 1) {
            fetchChannels();
        }
    }, []);

    return (
        channels.length > 0 ? <TvGuide channels={channels} /> : <div></div>
    );
}

const mapStateToProps = ({ channels }: StoreState): { channels: Channel[] } => {
    return { channels };
};

export default connect(mapStateToProps, { fetchChannels })(App);
