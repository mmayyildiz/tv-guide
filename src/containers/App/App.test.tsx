import React from "react";
import { shallow } from "enzyme";
import App from "./App";
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';


const mockStore = configureStore();
const store = mockStore();

const props = {
  "channels": [
    {
      "sid": "2002",
      "t": "BBC One Lon",
      "c": "101",
      "sf": "SD",
      "sg": "Entertainment"
    },
    {
      "sid": "2006",
      "t": "BBC Two",
      "c": "102",
      "sf": "SD",
      "sg": "Entertainment"
    },
  ]
};

const wrapper = shallow(
  <Provider store={store}>
    <App {...props} />
  </Provider>
);

describe('My App Component', () => {

  it('should render correctly with default props and redux store', () => {
    expect(wrapper).toMatchSnapshot();
  });

});

