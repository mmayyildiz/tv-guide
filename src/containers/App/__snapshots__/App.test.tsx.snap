// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My App Component should render correctly with default props and redux store 1`] = `ShallowWrapper {}`;
